### ISSUE TITLE
Specify scroll position when using hover

### ISSUE THREAD

**Issue by osadasami** [watir/watir#948]
When we use `el.hover`, Watir scrolls to the top of the element then hovers it.

But we don't always want to scroll to the top of the element when hovering.

Sometimes we need to scroll to the center or to the bottom of the elements when hovering.

it's impossible to do now.

If we use `el.scroll.to :center` and then `el.hover`, the scroll will go to the top of the element.

If we use `el.hover` and then `el.scroll.to :center` we will lose hover state.

I have found methods which use `scroll.to` inside:
- hover
- double_click
- right_click
- drag_and_drop_on
- drag_and_drop_by
- obscured?

So it would be great to specify scroll position in those methods.

Something like this:

```rb
def hover(scroll_position = :top)
  element_call(:wait_for_present) do
    scroll.to(scroll_position)
    driver.action.move_to(@element).perform
  end
end
```

**Comment by titusfortner** [comment]
Cool idea, would you like to make a PR for it?

**Comment by titusfortner** [comment]
Thanks for addressing with #949 
I'm adding more specs to ensure everything is the correct behavior, and tweaking a few things.

Let me document here the current browser behavior:
* Chromium element interaction
  * Element below viewport:  scrolls to *bottom*
  * Element above viewport scrolls to *top*
* Chromium using actions class
  * Element below viewport: scrolls to *bottom*
  * Element above viewport: scrolls to *bottom*
* Firefox element interaction
  * Element below viewport: scrolls to *bottom*
  * Element above viewport: scrolls to *bottom*
* Firefox using actions class
  * Element below viewport: *returns error*
  * Element above viewport: *returns error*

Firefox is correct in both of these behaviors.
Chromium [should always scroll to the bottom](https://bugs.chromium.org/p/chromedriver/issues/detail?id=3954)
Chromium [should not scroll when element out of bounds in Actions class](https://bugs.chromium.org/p/chromedriver/issues/detail?id=3955#makechanges)

`#obscured?` is designed to scroll as the browser would for a click action and see if anything is in the way, so it shouldn't take a parameter. It is currently always scrolling the element to the top of viewport, which is wrong, so it will be changed to check to see if the element is out of the viewport, then scroll it to the bottom if it is. This should make it much more accurate, and entirely accurate once the Chrome bug is fixed. It is not going to handle horizontal scrolling, which I'm ok with, but for reference this behavior could be improved if we want to do that. @jkotests does this work for you?

**Comment by titusfortner** [comment]
fixed and released in Watir 7.1
Thanks again!
